src/math/rem_pio2_large.rs

```diff
@@ -0,0 +1,489 @@
+///              for jk is 3,4,4,6 for single, double, extended, and quad.
+///              jk+1 must be 2 larger than you might expect so that our
+///              recomputation test works. (Up to 24 bits in the integer
+///              part (the 24 bits of it that we compute) and 23 bits in
+///              the fraction part may be lost to cancelation before we
+///              recompute.)
+///
+///      jz      local integer variable indicating the number of
+///              terms of ipio2[] used.
+///
+///      jx      nx - 1
+///
+///      jv      index for pointing to the suitable ipio2[] for the
+///              computation. In general, we want
+///                      ( 2^e0*x[0] * ipio2[jv-1]*2^(-24jv) )/8
+///              is an integer. Thus
+///                      e0-3-24*jv >= 0 or (e0-3)/24 >= jv
+///              Hence jv = max(0,(e0-3)/24).
+///
+///      jp      jp+1 is the number of terms in PIo2[] needed, jp = jk.
+///
+///      q[]     double array with integral value, representing the
+///              24-bits chunk of the product of x and 2/pi.
+///
+///      q0      the corresponding exponent of q[0]. Note that the
+///              exponent for q[i] would be q0-24*i.
+///
+///      PIo2[]  double precision array, obtained by cutting pi/2
+///              into 24 bits chunks.
+///
+///      f[]     ipio2[] in floating point
+///
+///      iq[]    integer array by breaking up q[] in 24-bits chunk.
+///
+///      fq[]    final product of x*(2/pi) in fq[0],..,fq[jk]
+///
+///      ih      integer. If >0 it indicates q[] is >= 0.5, hence
+///              it also indicates the *sign* of the result.
+
+const INIT_JK : [usize; 4] = [3,4,4,6]; /* initial value for jk */
+
+/// Table of constants for 2/pi, 396 Hex digits (476 decimal) of 2/pi
+///
+///              integer array, contains the (24*i)-th to (24*i+23)-th
+///              bit of 2/pi after binary point. The corresponding
+///              floating value is
+///
+///                      ipio2[i] * 2^(-24(i+1)).
+///
+/// NB: This table must have at least (e0-3)/24 + jk terms.
+///     For quad precision (e0 <= 16360, jk = 6), this is 686.
+#[cfg(not(ldbl_max_exp_more1024))]
+const IPIO2 : [i32; 66] = [
+    0xA2F983, 0x6E4E44, 0x1529FC, 0x2757D1, 0xF534DD, 0xC0DB62,
+    0x95993C, 0x439041, 0xFE5163, 0xABDEBB, 0xC561B7, 0x246E3A,
+    0x424DD2, 0xE00649, 0x2EEA09, 0xD1921C, 0xFE1DEB, 0x1CB129,
+    0xA73EE8, 0x8235F5, 0x2EBB44, 0x84E99C, 0x7026B4, 0x5F7E41,
+    0x3991D6, 0x398353, 0x39F49C, 0x845F8B, 0xBDF928, 0x3B1FF8,
+    0x97FFDE, 0x05980F, 0xEF2F11, 0x8B5A0A, 0x6D1F6D, 0x367ECF,
+    0x27CB09, 0xB74F46, 0x3F669E, 0x5FEA2D, 0x7527BA, 0xC7EBE5,
+    0xF17B3D, 0x0739F7, 0x8A5292, 0xEA6BFB, 0x5FB11F, 0x8D5D08,
+    0x560330, 0x46FC7B, 0x6BABF0, 0xCFBC20, 0x9AF436, 0x1DA9E3,
+    0x91615E, 0xE61B08, 0x659985, 0x5F14A0, 0x68408D, 0xFFD880,
+    0x4D7327, 0x310606, 0x1556CA, 0x73A8C9, 0x60E27B, 0xC08C6B,
+];
+
+#[cfg(ldbl_max_exp_more1024)]
```

Comment:
This table will always be ignored by the compiler because `--cfg ldbl_max_exp_more1024` is never passed to rustc. I think we want to change the contents of this table depending on the architecture. At least that's what the MUSL code does:

``` console
$ rg 'define.*LDBL_MAX_EXP'
arch/x32/bits/float.h
14:#define LDBL_MAX_EXP 16384

arch/sh/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/powerpc64/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/powerpc/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/or1k/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/mipsn32/bits/float.h
10:#define LDBL_MAX_EXP 16384

arch/microblaze/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/mips64/bits/float.h
10:#define LDBL_MAX_EXP 16384

arch/i386/bits/float.h
14:#define LDBL_MAX_EXP 16384

arch/mips/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/arm/bits/float.h
10:#define LDBL_MAX_EXP 1024

arch/aarch64/bits/float.h
10:#define LDBL_MAX_EXP 16384

arch/s390x/bits/float.h
10:#define LDBL_MAX_EXP 16384

arch/x86_64/bits/float.h
14:#define LDBL_MAX_EXP 16384
```

It seems that the smaller table is for 32-bit architectures and the bigger one is for 64-bit architectures. So you probably want to use `#[cfg(target_pointer_width = "32")]` on the smaller table and  `#[cfg(target_pointer_width = "64")]` on the bigger one.

Reply:
I believe this is actually based on `long double` sizes, where it's larger for architectures that support 80-bit or 128-bit floats. I don't think Rust supports larger than `f64`, so I don't think we need these.